src/model/utils/selection-post-fixer.js

```diff
@@ -197,13 +202,23 @@ function tryFixingNonCollapsedRage( range, schema ) {
 	// At this point we eliminated valid positions on text nodes so if one of range positions is placed inside a limit element
 	// then the range crossed limit element boundaries and needs to be fixed.
 	if ( isStartInLimit || isEndInLimit ) {
+		const bothInSameParent = ( !!start.nodeAfter && !!end.nodeBefore ) && start.nodeAfter === end.nodeBefore;
```

Comment:
I think we can omit `!!`.

Reply:
Also, I am not sure if the name is correct. It's more that the selection is on the same element, isn't it?

Reply:
The name is correct but the check is not. Updated to look for parent check.